homework_18/Box.java

```diff
@@ -0,0 +1,42 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Box<T extends Fruit> {
+    private final List<T> fruitList = new ArrayList<>();
+
+    public void add(T fruit) {
+        Objects.requireNonNull(fruit);
+        fruitList.add(fruit);
+    }
+
+    public void add(List<T> fruit) {
+        fruitList.addAll(fruit);
+    }
+
+    public double getWeight() {
+        if (fruitList.isEmpty()) {
+            return 0.0F;
```

Comment:
Again, why are you using float here? You've set `weight` as a double, so use double